SU-HSDO/suhumsci
# STN-593 :  Create Caption/Credit view mode and styles

# READY FOR REVIEW

## Summary
* Adds a new view mode with a style for captions that utilizes the Caption/Credit field associated with Media. 
* In accordance with the Implementation Details in [STN-592](https://sparkbox.atlassian.net/browse/STN-592) the priority is for the caption/credit field to work on banners, carousels, and hero images.
* Max character length is 120 characters
* Configuration changes to banners, heros, postcards, and views are not included as a means to protect the integrity of current sites.

See both [STN-592](https://sparkbox.atlassian.net/browse/STN-592) and [STN-593](https://sparkbox.atlassian.net/browse/STN-593) for details.

## Need Review By (Date)
11/9/2020

## Urgency
medium

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. The work in this PR includes a new view mode. You will probably need to run `lando drush @sparkbox_sandbox.local config-import --partial` to update your config so that you can use the new caption/credit view mode.
3. Hide the caption/credit label and trim characters.
    * Go to Structure > Media types > Image > Manage Display
    * Scroll down to the Field section
    * Find Caption/Credit
    * Under Label, change it from `Above` to `-Hidden-`
    * Under Format, change to `Trimmed` and set a limit of 120 characters
    * Save 
4. Update banners to display image caption/credit
    * Go to Structure > Paragraph Types
    * Find `Banner` and select to `Manage Display`
    * You should now be on Structure > Paragraph types > Hs banner > Manager display
    * Click on the settings gear icon to the RHS of the page under `Show row weights`
    * Update `View mode` from `Default` to `Caption/Credit`
    * Click on the update button
5. Create a basic page and add a banner in the hero area. Save and view. If you do not see image caption/credits you may need to add them for the image. 
    * Click on the contextual link to edit the image (There are 3 contextual links. It may take a couple tries before finding the correct one. :wink:)
    * Add a caption/credit, save, and return to the page
    * You can also update image caption/credits by going to Content > Media and searching for the image you want to add a caption/credit for.

    <img width="1280" alt="add caption" src="https://user-images.githubusercontent.com/12678977/98297830-07fb0400-1f83-11eb-8a10-d6aa50bfcd69.png">
6. Repeat the steps 4 and 5 above for hero and postcard paragraph types. Since Carousels are built off of Heros, once the paragraph type for heroes is updated, carousels will also be updated.
7. Test the caption/credit on views
    * Go to http://sparkbox-sandbox.suhumsci.loc/news
    * Click on the contextual link to edit the view
    * Under Fields: click on Content: New Image
    * Under View mode: select Caption/Credit and Apply
    * Save View
_You may need to add captions for images in views as well._
